dask_cloudprovider/hetzner/vserver.py

```diff
@@ -0,0 +1,176 @@
+        self.cluster._log(f"Created Hetzner vServer {self.name}")
+
+        return self.server.public_net.ipv4.ip
+
+    async def destroy_vm(self):
+        await self.cluster.call_async(self.client.servers.delete, server=self.server)
+        self.cluster._log(f"Terminated vServer {self.name}")
+
+
+class HetznerScheduler(SchedulerMixin, VServer):
+    """Scheduler running on a Hetzner server."""
+
+
+class HetznerWorker(WorkerMixin, VServer):
+    """Worker running on a Hetzner server."""
+
+
+class HetznerCluster(VMCluster):
+    """Cluster running on Hetzner cloud vServers.
+
+    VMs in Hetzner are referred to as vServers. This cluster manager constructs a Dask cluster
+    running on VMs.
+
+    When configuring your cluster you may find it useful to install the ``hcloud`` tool for querying the
+    Hetzner API for available options.
+
+    https://github.com/hetznercloud/cli
+
+    Parameters
+    ----------
+    image: str
+        The image to use for the host OS. This should be a Ubuntu variant.
+        You can list available images with ``hcloud image list|grep Ubuntu``.
+    location: str
+        The Hetzner location to launch you cluster in. A full list can be obtained with ``hcloud location list``.
+    server_type: str
+        The VM server type. You can get a full list with ``hcloud server-type list``.
+        The default is ``cx11`` which is vServer with 2GB RAM and 1 vCPU.
+    n_workers: int
+        Number of workers to initialise the cluster with. Defaults to ``0``.
+    worker_module: str
+        The Python module to run for the worker. Defaults to ``distributed.cli.dask_worker``
+    worker_options: dict
+        Params to be passed to the worker class.
+        See :class:`distributed.worker.Worker` for default worker class.
+        If you set ``worker_module`` then refer to the docstring for the custom worker class.
+    scheduler_options: dict
+        Params to be passed to the scheduler class.
+        See :class:`distributed.scheduler.Scheduler`.
+    env_vars: dict
+        Environment variables to be passed to the worker.
+
+    Example
+    --------
+
+    >>> from dask_cloudprovider.hetzner import HetznerCluster
+    >>> cluster = HetznerCluster(n_workers=1)
+
+    >>> from dask.distributed import Client
+    >>> client = Client(cluster)
+
+    >>> import dask.array as da
+    >>> arr = da.random.random((1000, 1000), chunks=(100, 100))
+    >>> arr.mean().compute()
+
+    >>> client.close()
+    >>> cluster.close()
+
+    """
+
+    def __init__(
+        self,
+        bootstrap: str = None,
+        image: str = None,
+        location: str = None,
+        server_type: str = None,
+        docker_image: str = None,
+        **kwargs,
+    ):
+        self.config = dask.config.get("cloudprovider.hetzner", {})
+
+        self.scheduler_class = HetznerScheduler
+        self.worker_class = HetznerWorker
+
+        self.image = image if image is not None else self.config.get("image")
```

Comment:
A couple of Dask releases ago we updated the `dask.config.get` function with an `override_with` kwarg to get rid of this less readable `var if var is not None else config`.

So this could be replaced with 

```suggestion
        self.image = dask.config.get("cloudprovider.hetzner.image", override_with=image)
```

I haven't gotten around to updating the other cluster managers yet but this could be a good place to start.